core/src/test/scala/dagr/core/execsystem/TaskManagerTest.scala

```diff
@@ -536,12 +538,15 @@ class TaskManagerTest extends UnitSpec with PrivateMethodTester with OptionValue
     taskManager.addTask(predecessor)
     taskManager.graphNodeStateFor(predecessor).value should be(GraphNodeState.PREDECESSORS_AND_UNEXPANDED)
     taskManager.stepExecution()
+    taskManager.joinOnRunningTasks(1000)
```

Comment:
we should also make sure that tests will fail if we expected all tasks to complete.